File: dp_labs_and_solutions/dp_labs_and_solutions/dp_java_labs_and_solutions/visitor/solution/SingleJourney.java
// SingleJourney.java

public class SingleJourney extends Journey
{
	public SingleJourney(String aAddress)
	{
		if (aAddress == null || aAddress.equals(""))
		{
			throw new DpException("SingleJourney.SingleJourney(String aAddress) -- aAddress == null || aAddress.equals(\"\").");
		}
		myAddress = aAddress;
	}

	public void accept(JourneyVisitor aJourneyVisitor)
	{
		if (aJourneyVisitor == null)
		{
			throw new DpException("void SingleJourney.accept(JourneyVisitor aJourneyVisitor) -- aJourneyVisitor == null.");
		}

		aJourneyVisitor.visit(this);
	}

	public String getAddress()
	{
		return myAddress;
	}

	private String myAddress;
}

File: dp_labs_and_solutions/original/dp_labs_and_solutions/dp_cpp_labs_and_solutions/visitor/solution/single_journey.cpp
// single_journey.cpp

#include <string>
#include "dp_exception.h"
#include "journey_visitor.h"
#include "single_journey.h"

single_journey::single_journey(const std::string & a_address)
{
	if (a_address == "")
	{
		throw dp_exception("single_journey::single_journey(const std::string & a_address) -- a_address == \"\".");
	}
	my_address = a_address;
}

void single_journey::accept(journey_visitor & a_journey_visitor)
{
	a_journey_visitor.visit(*this);
}

const std::string & single_journey::get_address() const
{
	return my_address;
}
